### PR TITLE
json-schema: v6 and v7 changed to better comply with the draft definitions.

### PR DESCRIPTION
**Pull request by afinch7**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://json-schema.org/draft-06/schema#
https://json-schema.org/draft-07/schema#
```
"type": ["object", "boolean"]
```

### CODE REVIEW

**Comment by cyrilletuzi** [comment]
@afinch7 What is the motivation of this PR? I don't see what it adds except refactoring, and I'm not confident in validating the changes, change from interface to type could have side effects.

**Comment by rolandjitsu** [comment]
@cyrilletuzi I second that. I cannot see how a schema object can be a boolean, at least not the root schema object.

**Comment by afinch7** [comment]
According to the draft 07 and 06 specs any field that can be a schema object can be a boolean as well.

Note this declaration in the 07 and 06 schemas. (see http://json-schema.org/draft-07/schema# and http://json-schema.org/draft-06/schema#)
```
"type": ["object", "boolean"]
```

This means that a schema like this
```
{
  "type": "object",
  "properties": {
    "aProp": false
  }
}
```
while mostly pointless is still valid, and more importantly
```
{
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "aProp": false
  }
}
```
is valid. 

I could have changed the definition of additionalProperties to be "boolean | JSONSchema6/7", but this doesn't correctly reflect the schema definition.

TL;DR;
The "additionalProperties" property was not functioning correctly before. Allowing for Schemas to be bools as well should better reflect the underlying spec, and fix this issue.

**Comment by afinch7** [comment]
@cyrilletuzi As for the interface to type change. I do see the potential for problems.
I.E.
```
class SomeClass implements JSONSchema7 
```
This will no longer compile.

I see there being two options to resolve this. Make this a breaking change or refactor this change to something like
```
type JSONSchemaSomething7 = JSONSchema7  | boolean;

interface JSONSchema7 {
  properties: JSONSchemaSomething7;
}
```
There may be a better way to refactor that not sure. What do you think?

**Comment by rolandjitsu** [comment]
@afinch7 I'm aware of that, but making the root schema a boolean makes no sense, especially when you're extending the interface or implementing it.

Your second suggestion would work though.

**Comment by afinch7** [comment]
Alright I created a test for the class implementation cases and 
 renamed my new type to `JSONSchema6Definition` and `JSONSchema7Definition`.
afinch7@0d7ab87136520634baf79ac5f4d43293966901ab

**Review by rolandjitsu — APPROVED** [review]
LGTM
